joomla/joomla-cms
# [4.1] [PHP 8.1] Fix PHP 8.1 deprecation notice for Rules Field

Pull Request for Issue # none

### Summary of Changes

Fixes `Deprecated: Automatic conversion of false to array is deprecated in /home/beat/www/4.0/libraries/src/Form/Field/RulesField.php on line 182`

In cases where an access.xml file isn't present for an extension, in Joomla 3.10 (any PHP version) as well as in Joomla 4.0 up to PHP 8.0.x, the permissions rules field can be used without issues. With PHP 8.1, when the file is not present, `Access::getActionsFromFile()` on line 171 returns false instead of an empty array. When adding elements, the boolean is converted to array at previous line 181, which raises that deprecation notice.

This PR fixes this, converting to array that `$this->actions` property which should anyway always be of type array.

I have discovered this issue testing permissions of a CB add-on, that uses _com_comprofiler.plugin.pluginname_ as asset name. And obviously, there is no access.xml file for that field.

### Testing Instructions

Source-code review should be enough here.

But to test, just go to permissions of Joomla to check everything works as before.

### Actual result BEFORE applying this Pull Request

Works.

### Expected result AFTER applying this Pull Request

Still works.

### Documentation Changes Required

None.
